src/dataflow/src/operator.rs

```diff
@@ -134,6 +135,21 @@ where
     where
         E: Data,
         L: FnMut(&D1) -> Result<bool, E> + 'static;
+
+    fn explode_fallible<D2, E, R2, I, L>(
```

Comment:
Can you add a doc comment on this that's analogous to the doc comments on the other methods in this trait?

Reply:
Gotta run to a meeting so this thought isn't totally finished but... why does this method have to exist? Can't you just fallible flat map to (row, diff) and then use the normal explode operator?

Reply:
Sure, but I don't see anything wrong with adding here if it makes the "fallible" API richer/more complete. I can get rid of it if you prefer.

Reply:
Random thought, hopefully not disruptive: I think the `fallible` ops could be consolidated under something like `explode` if we'd like. Each of `map`, `filter`, and `flatmap` are instances of it. IIRC @benesch implemented each separately to mirror timely, but timely mainly implemented them separately as a teaching mechanism (e.g. "here is the implementation of `map`"). If we want to draw down the number of these we def can; if we have any perf concerns, Rust + LLVM will compile closures passed in to more complicated operators with no overhead, whereas chaining two timely operators will have unrecoverable overhead.

Reply:
Yeah, that sounds good to me. I think this is fine for now then.